### PR TITLE
Move inference_hook from vernacentries to lemmas

### PR DESCRIPTION
**Pull request by SkySkimmer**
This hook handles obligations from the type of a Lemma.

Moving it to lemmas simplifies invocations of start_lemma_com.

### CODE REVIEW

File: vernac/lemmas.ml [hunk 1]
```diff
@@ -193,16 +193,38 @@ let start_lemma_with_initialization ?hook ~poly ~scope ~kind ~udecl sigma recgua
         | None -> p
         | Some tac -> pi1 @@ Proof.run_tactic Global.(env ()) tac p)) lemma
 
-let start_lemma_com ~program_mode ~poly ~scope ~kind ?inference_hook ?hook thms =
+let obligations_default_tac = ref (Proofview.tclUNIT)
+
+let inference_hook ~program_mode =
+  if program_mode then
```
Comment (ejgallego) [inline vernac/lemmas.ml]: why not remove the if here and push the logic to `start_lemma_com`?

Comment (SkySkimmer) [reply vernac/lemmas.ml]: why change it?

**Review by ejgallego — COMMENTED** [review]
The gain is small but OK, maybe we should not move the `default_tactic` hook, but keep the parameter and pass the tactic as argument [not the hook].

File: vernac/lemmas.ml (hunk 1 above)
Comment (ejgallego) [reply vernac/lemmas.ml]: It seems strange to have a conditional hook, I'd rather don't set the hook at all if not needed.

**Review by ejgallego — APPROVED** [review]
Indeed we don't gain much as noted by Gaëtan, but we can merge this; eventually we may move the "Command" parts in `Lemmas` to their own `ComTheorem` file, as done for other commands.

I'd suggest removing the logic from inference_hook as noted in the comments.

**Comment by ejgallego** [comment]
I'll merge when the overlay is pushed.

**Comment by SkySkimmer** [comment]
Changed my approach to move start_proof_com to vernacentries, this way no need for an overlay. We can do ComTheorem later.

**Pull request merged**